src/jbi/services.py

```diff
@@ -36,17 +47,35 @@ def bugzilla_check_health() -> ServiceHealth:
     return health
 
 
-def jira_check_health() -> ServiceHealth:
+def jira_check_health(actions: Actions) -> ServiceHealth:
     """Check health for Jira Service"""
     jira = get_jira()
     server_info = jira.get_server_info(True)
-    health: ServiceHealth = {"up": server_info is not None}
+    health: ServiceHealth = {
+        "up": server_info is not None,
+    }
+    if server_info is None:
+        return health
+
+    visible_projects = {project["key"] for project in jira_visible_projects(jira)}
+    configured_projects = {
+        action.parameters["jira_project_key"]
+        for action in actions.__root__
+        if "jira_project_key" in action.parameters
+    }
+    missing_projects = configured_projects - visible_projects
+    if missing_projects:
+        logger.error(
+            "Projects %s are not visible with configured credentials", missing_projects
+        )
+
+    health["all_projects_are_visible"] = not missing_projects
```

Comment:
It _almost_ feels like we should abstract the above into a separate function, like:
```python
health["all_projects_are_visible"] = jira_configured_projects_available(actions)
```
where `jira_configured_projects_available` is:
```python
def jira_configured_projects_available(actions: Actions) -> bool:
    jira = get_jira()
    visible_projects = {project["key"] for project in jira_visible_projects(jira)}
    missing_projects = actions.configured_jira_projects - visible_projects
    if missing_projects:
        logger.error(
            "Projects %s are not visible with configured credentials",
            missing_projects,
        )
    return not missing_projects
```

I don't feel particularly strongly about this, just trying things out.

Reply:
I like it ;)